### PR TITLE
Adding backwards compatibility for server cli

### PR DESCRIPTION
**Pull request by corey-nm**
This adds support for using the old version of deepsparse.server cli with a deprecation message:

```
deepsparse.server --task qa
deepsparse.server task qa
deepsparse.server --config_file /some/path
deepsparse.server config /some/path
```

Test plan:
- Since this is not replacing the new way of the api, none of the QA or unit tests need to be updated
- Unit tests
- Manual cli testing

### CODE REVIEW

File: src/deepsparse/server/cli.py [hunk 1]
```diff
@@ -166,30 +106,185 @@ def config(
...
+    log_level: str,
+    hot_reload_config: bool,
+    model_path: str,
+    batch_size: int,
+    num_cores: int,
+    num_workers: int,
+    integration: str,
+):
+    """
+    Start a DeepSparse inference server for serving the models and pipelines.
```
Comment (bfineran) [inline src/deepsparse/server/cli.py:165]: let's update help text to include the pathway w/o the named args

**Review by bfineran — COMMENTED** [review]
LGTM overall, as a follow up let's get ahead on identifying readmes that use the `task`/`config` pathways, there shouldn't be too many

File: src/deepsparse/server/cli.py (hunk 1 above)
Comment (corey-nm) [reply src/deepsparse/server/cli.py:165]: Updated

**Review by bfineran — APPROVED** [review]

File: src/deepsparse/server/cli.py [hunk 2]
```diff
@@ -166,30 +106,185 @@ def config(
...
+)
+@HOST_OPTION
+@PORT_OPTION
+@LOG_LEVEL_OPTION
+@HOT_RELOAD_OPTION
+@MODEL_OPTION
+@BATCH_OPTION
+@CORES_OPTION
+@WORKERS_OPTION
+@INTEGRATION_OPTION
```
Comment (rahul-tuli) [inline src/deepsparse/server/cli.py:148]: nice

**Review by rahul-tuli — APPROVED** [review]

**Pull request merged**